components/net/http_loader.rs

```diff
@@ -1103,7 +1079,7 @@ fn http_network_fetch(request: &Request,
     // do not. Once we support other kinds of fetches we'll need to be more fine grained here
     // since things like image fetches are classified differently by devtools
     let is_xhr = request.destination == Destination::None;
-    let wrapped_response = obtain_response(&factory, &url, &request.method,
+    let wrapped_response = obtain_response(context.connector.clone(), &url, &request.method,
```

Comment:
That was what I tried originally, and Rust's type checker became _so confused_ that I didn't want to try to debug. Please don't make me investigate it; it ends up trying to use the closure connector feature instead and it's not worth it.